src/middlewares/authenticationMiddleware.ts

```diff
@@ -0,0 +1,46 @@
+import jwt from 'jsonwebtoken'
+import { Response, Request, NextFunction } from 'express'
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/no-namespace
+  namespace Express {
+    // eslint-disable-next-line no-shadow
+    interface Request {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      userDetails: any
+    }
+  }
+}
+
+// eslint-disable-next-line import/prefer-default-export
+export const isAuthenticated = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  try {
+    if (!req.headers.authorization) {
+      return res.status(401).json({ message: 'Please Login' })
+    }
+    const token = req.headers.authorization?.split(' ')[1] || ''
+
+    if (!token) {
+      return res.status(401).json({ message: 'no access token found' })
+    }
+
+    jwt.verify(
+      token,
+      process.env.JWT_SECRET || 'USER-AUTH',
```

Comment:
```suggestion
      JWT_SECRET
```
JWT_SECRET can be imported from the `src/config`, so th dat whenever name change  for env you don't have change it multiple place